wasm/jit/jit_amd64.go

```diff
@@ -781,6 +783,129 @@ func (c *amd64Compiler) compileSub(o *wazeroir.OperationSub) error {
 	return nil
 }
 
+// compileMul adds instructions to multiply two operands which may be on the stack or registers.
+// After execution, the result of multiplication pushed onto the stack.
```

Comment:
that the prior RDX is pushed on the stack.. is that intuitive enough to assume as being a result of multiplication?

Reply:
actually RDX is not pushed on the stack, RDX holds the overflow info according to the manual, but we don't need to use it in Wasm convention